RocketChat/Rocket.Chat
# fix: Thread message preview

## Proposed changes (including videos or screenshots)


The thread preview message was not displaying emojis and when the parent message was a quoted message it was not displaying the correct message.

To fix the issues I did:

- Emoji Issue:  I used the `<GazzodownText /> ` component to call `<PreviewMarkup />` conditionally when passed the prop `preview`. The  `<GazzodownText /> ` component it's necessary because it has the `MarkupInteractionContext.Provider` that has the context necessary to display the emoji and other elements
- Empty parent message: Since the parent message it's a quote, the initial string from a quote message it's a link that references the original message followed by `\n` break line. I added a validation to check if it's a quoteAttachment message, if it is I remove the first markdown token responsible by the link

**Before:**
![image](https://user-images.githubusercontent.com/20212776/225731429-98e2b5e2-9aff-4f17-9178-da9ff3257c2f.png)

**After:**
![image](https://user-images.githubusercontent.com/20212776/225725000-9c4e9d3b-f8d0-4dec-9091-1ffa091406d0.png)





## Issue(s)


## Steps to test or reproduce

1. Open https://open.rocket.chat
2. Send a message
3. Quote the previous message
4. Send another sequential message 
5. Reply in thread the message sent on step 3
6. Write the reply message inside the thread and select the checkbox `Also send to channel`
7. Check it how it looks the Thread Message Preview in the Message List

## Further comments


Little improvements made: 
- Memoized `<ThreadMessagePreview />` and `<ThreadMessagePreviewBody />` components to avoid component re-renders  

NOTE: The extra spaces saw in the _Before_ image it's being fixed in a different PR (https://github.com/RocketChat/Rocket.Chat/pull/28434) 


TC-550
